frontend/interfaces/form_field.ts

```diff
@@ -12,9 +12,9 @@ export default PropTypes.shape({
   ]),
 });
 
-export interface IFormField {
+export interface IFormField<T = any[] | boolean | number | string> {
   error: string;
   name: string;
   onChange: (value: any) => void;
-  value: any[] | boolean | number | string;
+  value: T;
```

Comment:
why the generic here?

Reply:
it allows us to specify more restrictive typing for a field if desired, for example in `IQueryFormFields,` we can now specify `string` only: `name: IFormField<string>;`